src/Bicep.Core.UnitTests/Registry/MockRegistryBlobClient.cs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure;
+using Azure.Core;
+using Bicep.Core.Registry.Oci;
+using Bicep.Core.RegistryClient;
+using Moq;
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Bicep.Core.UnitTests.Registry
+{
+    /// <summary>
+    /// Mock OCI registry blob client. This client is intended to represent a single repository within a specific registry Uri.
+    /// </summary>
+    public class MockRegistryBlobClient: BicepRegistryBlobClient
+    {
+        private static MockRepository Repository = new(MockBehavior.Strict);
+
+        public MockRegistryBlobClient() : base()
+        {
+            // ensure we call the base parameterless constructor to prevent outgoing calls
+        }
+
+        // maps digest to blob bytes
+        public ConcurrentDictionary<string, ImmutableArray<byte>> Blobs { get; } = new();
+
+        // maps digest to manifest bytes
+        public ConcurrentDictionary<string, ImmutableArray<byte>> Manifests { get; } = new();
+
+        // maps tag to manifest digest
+        public ConcurrentDictionary<string, string> ManifestTags { get; } = new();
+
+        public override async Task<Response<DownloadBlobResult>> DownloadBlobAsync(string digest, DownloadBlobOptions? options = default, CancellationToken cancellationToken = default)
+        {
+            await Task.Yield();
```

Comment:
>Yield [](http://example.com/codeflow?start=23&length=5)

Intentionally doing this to ensure some sort of continuation happens to better match the behavior of the real client.